props/saber_fett263_buttons.h

```diff
@@ -285,6 +445,215 @@ EFFECT(faston); // for EFFECT_FAST_ON
+public:
+  virtual ReadStatus Read(FileReader* f) override {
+    int file;
+    if (!f || !f->IsOpen()) return ReadStatus::READ_FAIL;
+    memset(clash_rec, 0, sizeof(clash_rec));
+    STDOUT.println("Reading choreo.ini");
+    for (size_t i = 0; i < NELEM(clash_rec); i++) {
+      char action[33];
+      f->skipwhite();
+      f->readVariable(action);
+      if (!strcmp(action, "clash")) {
+          if (SFX_clsh) {
+             file = SFX_clsh.get_min_file();
+          } else {
+             file = SFX_clash.get_min_file();
+          }
+        clash_rec[i].stance = STANCE_CLASH;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        continue;
+      }
+      if (!strcmp(action, "lockup")) {
+          if (SFX_lock) {
+            file = SFX_lock.get_min_file();
+          } else {
+            file = SFX_lockup.get_min_file();
+          }
+        clash_rec[i].stance = STANCE_LOCKUP;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - SFX_bgnlock.get_min_file());
+        f->skipspace();
+        clash_rec[i].sound_number2 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        clash_rec[i].sound_number3 = (int)(f->readFloatValue() - SFX_endlock.get_min_file());
+        f->skipspace();
+        continue;
+      }
+      clash_rec[i].stance = STANCE_END;
+      return ReadStatus:: READ_END;
+      break;
+    }
+  }
+
+  void SetVariable(const char* variable, float v) override {
+  }
+
+  enum FormStance {
+    STANCE_END = 0,
+    STANCE_CLASH,
+    STANCE_LOCKUP,
+  };
+
+  struct FormEntry {
+    FormStance stance;
+    int sound_number1;
+    int sound_number2;
+    int sound_number3;
+  };
+
+  FormEntry clash_rec[32];
+};
+#endif
+
+#ifdef FETT263_EDIT_MODE_MENU
+// Edit Length 
+int length_edit_length = 0;
+
+// Edit Length Preview for Edit Mode
+// LengthEdit uses blade color and creates single white pixel at last pixel, black above last pixel
+template<class BASE, class LIGHTUP, class BLACK = BLACK>
+class LengthEdit {
+public:
+  void run(BladeBase* blade) {
+    base_.run(blade);
+    lightup_.run(blade);
+    black_.run(blade);
+    led_ = length_edit_length;
+  }
+
+  OverDriveColor getColor(int led) {
+    if (led == led_) return lightup_.getColor(led);
+    if (led > led_) return black_.getColor(led);
+    return base_.getColor(led);
+  }
+private:
+  int led_;
+  BASE base_;
+  LIGHTUP lightup_;
+  BLACK black_;
+};
+
+// Edit Style Settings
+
+class IntEdit {
+  public:
+    void run(BladeBase* blade) {  }
+    int getInteger(int led) { return int_edit_; }
+    static void SetIntValue(int value) { int_edit_ = value; }
+ private:
+   static int int_edit_;
+};
+
+int IntEdit::int_edit_ = 0;
+#endif
+  
+// Color List
+static constexpr Color16 color_list_[] = {
+    { 65535, 0, 0 }, // Red
+    { 65535, 3598, 0 }, // OrangeRed
+    { 65535, 17476, 0 }, // DarkOrange
```

Comment:
If we add a constexpr here:
https://github.com/profezzorn/ProffieOS/blob/8d051b93ce161cebb2ec1175738880d464a97ad2/styles/rgb.h#L28

Then we may be able to replace
    { 65535,0,0}, // Red 

with:
    Red::color(),

This would be better since we won't need to repeat the numbers, and we won't need the comment either.

Reply:
could you explain a bit more, not quite sure how to apply?

Reply:
I wasn't able to figure out yet so need a little help on how to add. 
Also I have several colors that I've perfected for the library that are used in the color list would I be able to add them as named colors to styles/color.h or should they stay Rgb16<>?
  { 7710, 15420, 51400 }, // IceBlue
  { 11102, 92, 53864 }, // Indigo
  { 24000, 0, 50536 }, // Purple
  { 30324, 0, 49768 }, // DeepPurple
  { 25700, 25700, 38550 }, // Silver
  { 21845, 21845, 51400 }, // Glacier
  { 46260, 46260, 65535 }, // IceWhite

Reply:
I added the constexpr in Rgb::color(), so now you should be able to replace {65536,0,0}, with Red::color(),   If that doesn't work, let me know.

For the colors that don't exist in my color list, just leave them as Rgb16.